clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -565,7 +640,7 @@ private void handleShareAcknowledgeFailure(Node fetchTarget,
                 TopicIdPartition tip = new TopicIdPartition(topic.topicId(),
                         partition.partitionIndex(),
                         metadata.topicNames().get(topic.topicId()));
-                metricsManager.recordFailedAcknowledgements(acknowledgeRequestState.getAcknowledgementsCount(tip));
+                metricsManager.recordFailedAcknowledgements(acknowledgeRequestState.getInFlightAcknowledgementsCount(tip));
                 acknowledgeRequestState.handleAcknowledgeErrorCode(tip, Errors.forException(error));
             }));
         } finally {
```

Comment:
I think we ought to call `acknowledgeRequestState.onFailedAttempt` or `acknowledgeRequestState.onSuccessfulAttempt` for every single code path that handles a response to an AcknowledgeRequestState. I think that's not true at the moment but it's hard to tell because in some cases we do it in the ShareConsumeRequestManager methods, and in some cases we do it in the AcknowledgeRequestState methods. Please can you make this consistent to make it easier to validate.

Reply:
Right yeah, now I have moved all the calls into the `handleShareAcknowledge` methods for consistency.